iis-common/src/test/java/eu/dnetlib/iis/common/java/io/SequenceFileTextValueReaderTest.java

```diff
@@ -0,0 +1,80 @@
+package eu.dnetlib.iis.common.java.io;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.isA;
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+
+import org.apache.hadoop.fs.Path;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
+
+import eu.dnetlib.iis.common.IntegrationTest;
+
+@Category(IntegrationTest.class)
+public class SequenceFileTextValueReaderTest {
+
+    private static final String SEQUENCE_FILES = "sequence_files";
+    private static final String ONE_FILE = SEQUENCE_FILES+"/part-00000";
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    //------------------------ TESTS --------------------------
+
+    @Test
+    public void readFile() throws Exception {
+        // given
+        SequenceFileTextValueReader reader = newReader(ONE_FILE);
+
+        // execute
+        ArrayList<String> items = readAll(reader);
+
+        // assert
+        assertThat(items, contains("a", "bb", "ccc"));
+    }
+
+    @Test
+    public void readDirectory() throws Exception {
+        // given
+        SequenceFileTextValueReader reader = newReader(SEQUENCE_FILES);
+
+        // execute
+        ArrayList<String> items = readAll(reader);
+
+        // assert
+        assertThat(items, contains("a", "bb", "ccc", "a", "bb", "ccc"));
+    }
+
+    @Test
+    public void closeReader() throws Exception {
+        // given
+        SequenceFileTextValueReader reader = newReader(ONE_FILE);
+
+        // execute
+        reader.close();
+
+        // assert
+        thrown.expectCause(isA(IOException.class));
+        reader.hasNext();
```

Comment:
You really test here that `hasNext` will fail after closing reader.
So i would name this test method `hasNext_closed_reader` or something similar.
Going further `reader.hasNext()` should be really in execute section (and `reader.close()` in given - because it is preparing the state of the service). But due to the way `ExpectedException` is constructed in junit, we cant really follow the rules of this 3 sections.
In conclusion I would exceptionally skip section comments in this particular test, as they are misleading.